psycop/common/feature_generation/sequences/patient_loaders.py

```diff
@@ -44,25 +45,42 @@ def format_diagnosis_columns(self, df: pl.LazyFrame) -> pl.LazyFrame:
             .explode("value")
             .with_columns(
                 [
-                    pl.col("value")
-                    .str.slice(offset=2)  # Remove prefix, e.g. A: or B: or Z:
-                    .str.strip()  # Replace trailing whitespaces
-                    .str.replace(
-                        "^D",
-                        "",
-                    ),  # In the diagnosis DF432, the D is only in the Danish system and doesn't carry meaning. Remove it.
-                    pl.col("value")
-                    .str.slice(offset=0, length=1)
-                    .alias("type"),  # Exctract the type, e.g. for A:DF2, extract A
+                    pl.col("value").str.split_exact(
+                        ":",
+                        1,
+                    ),  # Split diagnosis column into prefix (e.g., A, B, or Z) and diagnosis code
                     pl.lit("diagnosis").alias(
                         "source",
                     ),  # Add a source column, indicating diagnoses
                 ],
             )
-            .rename({"datotid_slut": "timestamp"})
+            .unnest("value")  # Unnest prefix and diagnosis code into separate columns
+            .with_columns(
+                pl.col("field_1")
+                .str.replace(
+                    "^D",
+                    "",
+                )
+                .str.strip(),
+            )  # In the diagnosis DF432, the D is only in the Danish system and doesn't carry meaning. Remove it.)
+            .rename(
+                {"datotid_slut": "timestamp", "field_0": "type", "field_1": "value"},
+            )
+        ).unique(maintain_order=True)
+
+        return df.drop_nulls()  # Drop all rows with no diagnosis
+
+
+def keep_if_min_n_visits(
+    event_df: pl.LazyFrame,
+    n_visits: Union[int, None],
+) -> pl.LazyFrame:
+    if n_visits:
```

Comment:
The if statement should not be in this function; why would we call the function, if we don't want it to filter? :-)